src/tagmanager/tm_workspace.c

```diff
@@ -746,9 +847,12 @@ static gint sort_found_tags(gconstpointer a, gconstpointer b, gpointer user_data
 	const TMTag *t1 = *((TMTag **) a);
 	const TMTag *t2 = *((TMTag **) b);
 
-	/* sort local vars first (with highest line number first), followed
-	 * by tags from current file, followed by workspace tags, followed by
-	 * global tags */
+	/* sort local vars first (with highest line number first),
```

Comment:
Please add this list to the manual.

When it says "local" vars, do you mean all locals in the current function, or only those declared before the cursor?

Reply:
IMO this is not something that should be documented. For users this whole thing is "geany does its best to place more relevant symbols to the top of the list" and how Geany does this is an implementation detail. You won't really notice this logic in practice, just that the symbol you want to type (hopefully) appears at or near the top of the list. If we happen to add support for LSP, each server will have its own logic which will be totally different from what we use in Geany now.

Reply:
> When it says "local" vars, do you mean all locals in the current function, or only those declared before the cursor?

Currently it's those before the cursor.

Reply:
> Currently it's those before the cursor.

Ok, its fine for C, if locals only and sorting only its ok for C++ so long as no other declarations (eg members) that happen after the cursor are not excluded from the autocomplete (even if they get sorted later).

Reply:
@elextr We have already been there :-)

https://github.com/geany/geany/pull/3175#issuecomment-1107821608

I haven't changed anything about this in this PR, it works as before.

Reply:
Never rely on my umm, you know thingy, umm, oh yeah ... memory!!